config/samples/xgboost-dist/utils.py

```
@@ -0,0 +1,276 @@
+
+    return x_start, x_end
+
+def dump_model(model, type, model_path, args):
+    """
+    dump the trained model into local place
+    you can update this function to store the model into a remote place
+    :param model:
+    :param path:
+    :return:
+    """
+    if model is None:
+        raise Exception("fail to get the xgboost train model")
+    else:
+        if type == "local":
+            joblib.dump(model, model_path)
+            logging.info("Dump model into local place %s", model_path)
+
+        elif type == "oss":
+
+            oss_param = parse_parameters(args.oss_param, ",", ":")
+            if oss_param is None:
+                raise Exception("Please input the oss config to store trained model")
+                return False
+
+            oss_param['path'] = args.model_path
+            oss_param['endpoint'] = 'http://oss-cn-hangzhou-zmf.aliyuncs.com'
+
+            dump_model_to_oss(oss_param, model)
+
+            logging.info("Dump model into oss place %s", args.model_path)
+
+        else:
+            logging.warning("other model storage is not supported")
+
+    return True
+
+def read_model(type, model_path, args):
+    """
+    read model from physical storage
+    :param type:
+    :param args:
+    :return:
+    """
+    model = None
+
+    if type == "local":
+        model = joblib.load(model_path)
+        logging.info("Read model from local place %s", model_path)
+
+    elif type == "oss":
+        oss_param = parse_parameters(args.oss_param, ",", ":")
+        if oss_param is None:
+            raise Exception("Please input the oss config to store trained model")
+            return False
+
+        oss_param['path'] = args.model_path
+        oss_param['endpoint'] = 'http://oss-cn-hangzhou-zmf.aliyuncs.com'
+
+        model = read_model_from_oss(oss_param)
+        logging.info("read model from oss place %s", model_path)
+
+    else:
+        logging.warning("the other model storage is not supported")
```

this should raise an exception instead of warning, right?